### ISSUE TITLE
[Windows] test_winconsoleio leaks references

### ISSUE THREAD

**Issue by vstinner** [python/cpython#74786]
BPO | [30601](https://bugs.python.org/issue30601)
--- | :---
Nosy | @pfmoore, @vstinner, @tjguk, @zware, @eryksun, @zooba, @segevfiner
PRs | <li>python/cpython#2003</li><li>python/cpython#2008</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2017-06-09.09:37:55.438>
created_at = <Date 2017-06-08.14:59:13.711>
labels = ['3.7', 'tests', 'OS-windows', 'performance']
title = '[Windows] test_winconsoleio leaks references'
updated_at = <Date 2017-06-09.09:37:55.436>
user = 'https://github.com/vstinner'
```

bugs.python.org fields:
```python
activity = <Date 2017-06-09.09:37:55.436>
actor = 'vstinner'
assignee = 'none'
closed = True
closed_date = <Date 2017-06-09.09:37:55.438>
closer = 'vstinner'
components = ['Tests', 'Windows']
creation = <Date 2017-06-08.14:59:13.711>
creator = 'vstinner'
dependencies = []
files = []
hgrepos = []
issue_num = 30601
keywords = []
message_count = 7.0
messages = ['295445', '295446', '295447', '295452', '295456', '295465', '295507']
nosy_count = 7.0
nosy_names = ['paul.moore', 'vstinner', 'tim.golden', 'zach.ware', 'eryksun', 'steve.dower', 'Segev Finer']
pr_nums = ['2003', '2008']
priority = 'normal'
resolution = 'fixed'
stage = 'resolved'
status = 'closed'
superseder = None
type = 'resource usage'
url = 'https://bugs.python.org/issue30601'
versions = ['Python 3.7']
```

</p></details>

**Comment by vstinner** [comment]
http://buildbot.python.org/all/builders/AMD64%20Windows8.1%20Refleaks%203.x/builds/22/steps/test/logs/stdio

0:00:12 [  7/405/1] test_winconsoleio failed
beginning 6 repetitions
123456
......
test_winconsoleio leaked [43, 43, 43] references, sum=129

**Comment by vstinner** [comment]
The following code leaks one reference:

```py
f = _io._WindowsConsoleIO('CONIO$', 'r')
f.close()
f = None
```

**Comment by vstinner** [comment]
With https://github.com/python/cpython/pull/2003 test_winconsoleio doesn't leak anymore.

**Comment by vstinner** [comment]
New changeset 29adc13bd797d9c9e7fcb893a7c49ce7f7ad388c by Victor Stinner in branch 'master':
bpo-30601: Fix a refleak in WindowsConsoleIO (bpo-2003)
https://github.com/python/cpython/commit/29adc13bd797d9c9e7fcb893a7c49ce7f7ad388c

**Comment by eryksun** [comment]
Segev included a fix for this leak in [PR 1927](https://github.com/python/cpython/pull/1927) for bpo-30555, but I suppose it's better to fix it separately.

**Comment by vstinner** [comment]
New changeset 31b950ab86f5c99a18c16dbf1900b854b94e0659 by Victor Stinner in branch '3.6':
bpo-30601: Fix a refleak in WindowsConsoleIO (bpo-2003) (bpo-2008)
https://github.com/python/cpython/commit/31b950ab86f5c99a18c16dbf1900b854b94e0659

**Comment by vstinner** [comment]
Eryk Sun: "Segev included a fix for this leak in [PR 1927](https://github.com/python/cpython/pull/1927) for bpo-30555, but I suppose it's better to fix it separately."

I saw this PR but it changes a lot of code. I didn't notice that it fixes this bug as well.

I chose to fix quickly the refleak to be able to detect more easily future refleaks (ie. repair the new refleak windows buildbot!

--

"AMD64 Windows8.1 Refleaks 3.6" buildbot confirms that test_winconsoleio doesn't leak anymore! So I close this issue.
